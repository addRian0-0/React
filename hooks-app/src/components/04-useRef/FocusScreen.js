import React, { useRef } from 'react';
import "../01-useState/counter.css";

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick = () => {
        inputRef.current.select()
    }

    return (
        <div>
            <h1>Focus Screen</h1>

            <input
                ref={inputRef}
                className="form-control"
                placeholder="addRian0-0"
            />

            <button
                className="btn btn-primary"
                onClick={handleClick}
            >
                Focus
            </button>

        </div>
    )
}