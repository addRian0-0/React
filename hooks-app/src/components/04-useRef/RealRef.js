import React, { useState } from 'react';
import { Multiple } from "../03-examples/Multiple";
import "../01-useState/counter.css";

export const RealRef = () => {

    const [show, setshow] = useState(false);

    return (
        <div>
            <h1>Real example useRef</h1>
            <hr />

            {show && <Multiple />}

            <button
                className="btn btn-primary"
                onClick={() => {
                    setshow(!show)
                }}
            >
                Show | Hide
            </button>

        </div>
    )
}
