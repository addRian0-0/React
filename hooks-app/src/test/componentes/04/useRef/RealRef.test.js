import { renderHook, act } from '@testing-library/react-hooks';
import { shallow } from "enzyme";
import { RealRef } from '../../../../components/04-useRef/RealRef';

describe('pruebas en realRef', () => {
    
    const wrapper = shallow( <RealRef/> );

    test('debe hacer el snapshot', () => {
       
        expect(wrapper).toMatchSnapshot();
        
    });

    test('debe mostrar el componente', () => {
        
        wrapper.find('button').simulate('click');
        expect(wrapper.find('Multiple').exists()).toBe(true);

    });

})
