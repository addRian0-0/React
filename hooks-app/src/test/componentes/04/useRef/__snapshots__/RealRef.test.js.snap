// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en realRef debe hacer el snapshot 1`] = `
<div>
  <h1>
    Real example useRef
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show | Hide
  </button>
</div>
`;
